stdlib/runtime/string.gr

```diff
@@ -37,21 +37,13 @@ primitive (&&): (Bool, Bool) -> Bool = "@and"
 primitive (||): (Bool, Bool) -> Bool = "@or"
 
 enum StringList {
-  SLEmpty,
-  SLCons(String, StringList),
-}
-
-@disableGC
-let slConsDisableGc = (a, b) => {
-  Memory.incRef(WasmI32.fromGrain(SLCons))
-  Memory.incRef(WasmI32.fromGrain(a))
-  // for easier chaining, we don't incRef `b`
-  SLCons(a, b)
+  [],
+  [...](String, StringList),
```

Comment:
This is 🔥  Love it

Reply:
Did we have an issue about this somewhere that should be closed by this PR?

Reply:
Probably https://github.com/grain-lang/grain/issues/552 but that's not _quite_ ready to be closed.